### PR TITLE
Initial import of proposed marketplace models

### PR DESCRIPTION
**Pull request by johnoliver**
# Checklist

- [ ] You added tests to cover the change
- [ ] `mvn clean install` build and test completes
- [ ] You changed or added to the documentation

### CODE REVIEW

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/HeapSize.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package net.adoptium.marketplace.schema;
+
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(type = SchemaType.STRING, enumeration = {"normal", "large"}, example = "normal")
```
Comment (karianna) [inline marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/HeapSize.java:6]: Not 100% sure on this but I think Large heap size concept has gone away permanently

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/JvmImpl.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package net.adoptium.marketplace.schema;
+
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+/*
+    Use a schema ref as adopt and adoptium will have a different subset of implementations
+*/
+@Schema(type = SchemaType.STRING, enumeration = {"hotspot", "openj9", "dragonwell"})
+public enum JvmImpl {
```
Comment (karianna) [inline marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/JvmImpl.java:10]: should we fix the impl vs vendor thing now?

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/ReleaseType.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package net.adoptium.marketplace.schema;
+
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(type = SchemaType.STRING, defaultValue = "ga", enumeration = {"ga", "ea"})
+public enum ReleaseType {
```
Comment (karianna) [inline marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/ReleaseType.java:7]: not sure if the marketplace is allowed to do EA builds

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package net.adoptium.marketplace.schema;
+
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+/*
+    Use a schema ref as adopt and adoptium will have a different subset of vendors
+*/
+public enum Vendor {
+    adoptium, adoptopenjdk, openjdk, alibaba, ibm, eclipse;
```
Comment (karianna) [inline marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java:9]: is it better to have adopt and eclipse as aliases of adoptium or some such?

File: marketplace/pom.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>net.adoptium.marketplace</groupId>
+    <artifactId>adoptium-marketplace-schema-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>9</maven.compiler.source>
```
Comment (karianna) [inline marketplace/pom.xml:12]: Java 9 seems an odd choice?

**Review by karianna — CHANGES_REQUESTED** [review]
Looks pretty good so far, not sure where the TCK and AQAVit passing checks should occur or be recorded.

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/HeapSize.java (hunk 1 above)
Comment (johnoliver) [reply marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/HeapSize.java:6]: If we are going to maintain backward compatibility for old builds, will need it. If not can get rid of it

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java (hunk 4 above)
Comment (johnoliver) [reply marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java:9]: can possibly get rid of eclipse, iirc I believe this is the former vendor used from openj9, however presumably the openj9 builds will not show in the market place, and semeru will have a vendor of ibm

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/HeapSize.java (hunk 1 above)
Comment (karianna) [reply marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/HeapSize.java:6]: I think you can drop it

File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java (hunk 4 above)
Comment (karianna) [reply marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java:9]: I'd possibly keep eclipse given the market confusion over eclipse vs adoptium vs temurin vs adopt.  Aliases will help here I think

Comment (karianna) [reply marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/Vendor.java:9]: after last night's discussion, drop eclipse

**Comment by karianna** [comment]
Could we not move this to Java 17 in keeping with rest of API?

**Comment by johnoliver** [comment]
PR change log:
- Removed ReleaseType
- Removed eclipse as a vendor
- Removed HeapSize
- Bumped to use Java 17
- Added Distribution

**Comment by tellison** [comment]
FWIW since this is now a big PR, I was going to wait until it landed as a WIP then create a number of smaller updates.

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
LGTM now - I am update main a little bit more with dependencies, but that's always a battle.

**Pull request merged**